File: UI_Frame_SC.lua
frameBaseOVL = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(2, 2, 1026, 128),
  onScreen = Rect:New(0, 642, 1024, 768),
  z = -2
})
ccPaneOVL = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(2, 130, 684, 173),
  onScreen = Rect:New(102, 650, 784, 693),
  z = -5
})
menuMag_Button = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(0, 669, 42, 755),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_Frame_OVL",
    source = Rect:New(596, 345, 633, 422),
    onScreen = Rect:New(0, 673, 37, 750)
  }),
  OnUp = function(this)
    this:Send("UIFrame", "Disable")
    Scene:BeginStream({
      stream = "MainMenuPopup",
      scene = "UI_Menu_SC",
      captureInput = true,
      inVP = false,
      save = false
    })
  end
})
menuKeyOpener = AR:Override({
  OnKeyDown = function(this, key)
    if key == Key.esc then
      menuMag_Button:OnUp()
      return true
    end
  end
})
menuFlashOVL = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(596, 345, 633, 422),
  onScreen = Rect:New(0, 673, 37, 750),
  z = 1,
  active = false
})
menuFlashAnim = AR:Override({
  active = false,
  alias = "MenuButton",
  durations = {loop = 0.5, full = 2},
  alphas = {on = 1, off = 0},
  RunOnce = function(this)
    menuFlashOVL.active = true
    this.elapsed = 0
  end,
  Run = function(this)
    menuFlashOVL.localAlpha = math.sinlerp2(this.alphas.off, this.alphas.on, this.elapsed / this.durations.loop)
    this.elapsed = this.elapsed + Scene.elapsedTime
    if this.elapsed >= this.durations.full then
      this:Done()
    end
  end,
  Receive = function(this, sender, action)
    if action == "Flash" then
      this:Restart()
    end
  end,
  OnDone = function(this)
    menuFlashOVL.active = false
  end
})
if not menuFlashAnim:RegisterAlias(menuFlashAnim.alias) then
  error(string.format("Could not register expected alias %s", Stringify(menuFlashAnim.alias)))
end
awardMenuOVL = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(596, 424, 633, 501),
  onScreen = Rect:New(0, 673, 37, 750),
  z = 1,
  active = false
})
awardFlashOVL = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(596, 503, 633, 580),
  onScreen = Rect:New(0, 673, 37, 750),
  z = 2,
  active = false
})
awardFlashSFX = AR:Sound({
  sounds = {
    "Cafe_Bell_SFX"
  },
  channel = "UIFX",
  volume = 0.85,
  active = false
})
awardFlashAnim = AR:Override({
  active = false,
  alias = "AwardMenuButton",
  durations = {loop = 1.2, full = 2.4},
  alphas = {on = 1, off = 0},
  RunOnce = function(this)
    awardMenuOVL.active = true
    awardFlashOVL.active = true
    this.elapsed = 0
  end,
  Run = function(this)
    awardFlashOVL.localAlpha = math.sinlerp2(this.alphas.off, this.alphas.on, this.elapsed / this.durations.loop)
    this.elapsed = this.elapsed + Scene.elapsedTime
    if this.elapsed >= this.durations.full then
      this:Done()
    end
  end,
  Receive = function(this, sender, action)
    if action == "Flash" then
      this:Restart()
    end
  end,
  OnDone = function(this)
    awardMenuOVL.active = false
    awardFlashOVL.active = false
  end
})
if not awardFlashAnim:RegisterAlias(awardFlashAnim.alias) then
  error(string.format("Could not register expected alias %s", Stringify(awardFlashAnim.alias)))
end
quitOvr = AR:Override({
  OnExitRequest = function()
    if Scene:IsStreamRunning("MainMenuPopup") then
      Scene:Change("MainMenuPopup", "UI_ToExit1_SC")
    else
      Scene:BeginStream({
        scene = "UI_ToExit1_SC",
        stream = "MainMenuPopup",
        captureInput = true,
        inVP = false,
        save = false
      })
    end
  end
})
uiFrame:Attach(frameBaseOVL, ccPaneOVL, menuMag_Button, menuKeyOpener, menuFlashOVL, menuFlashAnim, quitOvr)

File: Ciftree/UI_Frame_SC.js
function UI_Frame_SC() {
    let frameBaseOVL = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: (Flags.Touch) ? [2, 130, 1026, 128] : [2, 2, 1026, 128],
        onScreen: [0, 642, 1024, 768],
        z: 1
    });
    let ccPaneOVL = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [2, 258, 684, 173],
        onScreen: [102, 650, 784, 693],
        z: -5
    });
    let menuMag_Button = AR.Button({
        hs: AR.Hotspot({
            onScreen: [0, 669, 42, 755],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_Frame_OVL",
            source: [596, 473, 633, 550],
            onScreen: [0, 673, 37, 750],
            z: 1
        }),
        OnUp: function() {
            //alert(autotext.SequenceJR_01);
            uiFrame.classList.add("disabled");
            Scene.LetsGo("TitleMenu_SC");
        }
    });
    let menuFlashOVL = AR.Overlay({
        id: "menuFlashOVL",
        ovl: "UI_Frame_OVL",
        source: [596, 345, 633, 422],
        onScreen: [0, 673, 37, 750],
        z: 1,
        active: false
    });
    let awardMenuOVL = AR.Overlay({
        id: "awardMenuOVL",
        ovl: "UI_Frame_OVL",
        source: [596, 424, 633, 501],
        onScreen: [0, 673, 37, 750],
        z: 1,
        active: false
    });
    let awardFlashOVL = AR.Overlay({
        id: "awardFlashOVL",
        ovl: "UI_Frame_OVL",
        source: [596, 503, 633, 580],
        onScreen: [0, 673, 37, 750],
        z: 2,
        active: false
    });
    let touchNav = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [686, 359, 937, 426],
        onScreen: [691, 697, 760, 764],
        z: 1,
        active: Flags.Touch,
        id: "touchNav"
    });
    let touchNav_Button = AR.Button({
        hs: AR.Hotspot({
            ref: "UI_Frame_OVL",
            source: [686, 359, 937, 426],
            onScreen: [691, 697, 760, 764],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [686, 428, 937, 495],
          onScreen: [691, 697, 760, 764],
        }),
        z: 3,
        active: Flags.Touch,
        id: "touchNav_Button",
        OnUp: enableTouchNav.bind(this) 
    });
    let touchNavPressed = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [686, 497, 937, 564],
        onScreen: [691, 697, 760, 764],
        z: 1,
        active: false,
        id: "touchNavPressed"
    });
    let touchNavPressed_Button = AR.Button({
        hs: AR.Hotspot({
            ref: "UI_Frame_OVL",
            source: [686, 497, 937, 564],
            onScreen: [691, 697, 760, 764],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [686, 566, 937, 644],
          onScreen: [691, 697, 760, 764],
        }),
        z: 3,
        active: false,
        id: "touchNavPressed_Button",
        OnUp: disableTouchNav.bind(this) 
    });
    function enableTouchNav() {
        document.querySelector("#touchNav").style.visibility = "hidden";
        document.querySelector("#touchNav_Button").style.visibility = "hidden";
        document.querySelector("#touchNavPressed").style.visibility = "visible";
        document.querySelector("#touchNavPressed_Button").style.visibility = "visible";
        document.querySelector("body").classList.add("navHint");
    }
    function disableTouchNav() {
        document.querySelector("#touchNav").style.visibility = "visible";
        document.querySelector("#touchNav_Button").style.visibility = "visible";
        document.querySelector("#touchNavPressed").style.visibility = "hidden";
        document.querySelector("#touchNavPressed_Button").style.visibility = "hidden";
        document.querySelector("body").classList.remove("navHint");
    }
    document.querySelector("#uiFrame").append(frameBaseOVL, ccPaneOVL, menuMag_Button, menuFlashOVL, touchNav, touchNav_Button, touchNavPressed, touchNavPressed_Button);
}